File: sdk/src/main/java/com/sygic/travel/sdk/model/StResponse.java
package com.sygic.travel.sdk.model;

import com.google.gson.annotations.SerializedName;
import com.sygic.travel.sdk.model.media.Medium;
import com.sygic.travel.sdk.model.place.Detail;
import com.sygic.travel.sdk.model.place.Place;

import java.util.List;

/**
 * API response
 */
public class StResponse {
	public String status;
	public int statusCode;
	public String statusMessage;
	public Data data;

	public String getStatus() {
		return status;
	}

	public int getStatusCode() {
		return statusCode;
	}

	public String getStatusMessage() {
		return statusMessage;
	}

	public Data getData() {
		return data;
	}

	/**
	 * Contains data from the API response. Only one property should be assinged.
	 */
	public class Data {
		public static final String PLACE = "place";
		public static final String MEDIA = "media";
		public static final String PLACES = "places";

		@SerializedName(PLACES)
		private List<Place> places;

		@SerializedName(PLACE)
		private Detail detail;

		@SerializedName(MEDIA)
		private List<Medium> media;

		public List<Place> getPlaces() {
			return places;
		}

		public Detail getDetail() {
			return detail;
		}

		public List<Medium> getMedia() {
			return media;
		}
	}
}

File: sdk/src/main/java/com/sygic/travel/sdk/api/responseWrappers/StResponse.kt
package com.sygic.travel.sdk.api.responseWrappers

import com.google.gson.annotations.SerializedName

/**
 * Base of an API response. It's extended by specific respond classes containing the specific data.
 * It's abstract, because of the abstract [.getData] method.
 */
internal abstract class StResponse {
    @SerializedName("status_code")
    val statusCode: Int = 0

    @SerializedName("error")
    val error: Error? = null

    inner class Error {
        val id: String? = null
        val args: List<String>? = null
    }

    companion object {
        val STATUS_OK = 200
    }
}
